File: src/EnglishEasyRead.DictionaryApi/Tokenizer.fs
namespace EnglishEasyRead.DictionaryApi

module Tokenizer = 
    open System.Text.RegularExpressions
    
    let tokenize text = 
        Regex.Matches(text, @"([a-z]+)", RegexOptions.IgnoreCase)
        |> Seq.cast<Match>
        |> Seq.choose (fun m -> 
               if (m.Success) then Some(m.Groups.[1].Value)
               else None)
        |> Seq.map (fun w -> w.ToLower())

File: src/EnglishEasyRead.Dictionary/Tokenizer.cs
using System.Linq;
using System.Text.RegularExpressions;

namespace EnglishEasyRead.Dictionary
{
    public static class Tokenizer
    {
        public static string[] Tokenize(string text)
        {
            return Regex.Matches(text, @"([a-z]+)", RegexOptions.IgnoreCase)
                .Cast<Match>()
                .Where(m => m.Success)
                .Select(m => m.Groups[1].Value)
                .Select(t => t.ToLower())
                .ToArray();
        }
    }
}